### PR TITLE
feat(liteFullNodeTool): optimize liteFullNodeTool

### PR DESCRIPTION
**Pull request by halibobo1205**
**What does this PR do?**
    1.  disable split history data with light node data
    2.  disable merge history data which split from light node data
    3.  disable split snapshot if block is less than 65536
    4.  add latest 65536 blocks and related transactions
    5.  remove unnecessary check, minimum dbs for light node
**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

**Review by zhang0125 — APPROVED** [review]

File: framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java [hunk 1]
```diff
@@ -270,88 +249,50 @@ private long getLatestBlockHeaderNum(String databaseDir) throws IOException, Roc
...
-          destBlockDb.put(blockId, block);
-        });
+    long startIndex = latestBlockNum - RECENT_BLKS + 1;
+    // put the recent blocks and trans in snapshot
+    for (long blockNum = startIndex; blockNum <= latestBlockNum; blockNum++) {
+      try {
+        byte[] blockId = getDataFromSourceDB(sourceDir, BLOCK_INDEX_DB_NAME,
+            Longs.toByteArray(blockNum));
+        byte[] block = getDataFromSourceDB(sourceDir, BLOCK_DB_NAME, blockId);
+        // put  block
```
Comment (guoquanwu) [inline framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java:276]: two blank space

File: framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java [hunk 2]
```diff
@@ -270,88 +249,50 @@ private long getLatestBlockHeaderNum(String databaseDir) throws IOException, Roc
...
   /**
-   * Syncing block from peer that needs latest block and genesis block,
-   * also VM need recent blocks.
+   * recent blocks, trans and genesis block.
    */
-  private void fillSnapshotBlockDb(String sourceDir, String snapshotDir)
+  private void fillSnapshotBlockAndTransDb(String sourceDir, String snapshotDir)
           throws IOException, RocksDBException {
-    logger.info("-- begin to fill latest block and genesis block to snapshot");
+    logger.info("-- begin to fill {} block , genesis block and trans  to snapshot", RECENT_BLKS);
```
Comment (guoquanwu) [inline framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java:256]: two blank space

File: framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java [hunk 3]
```diff
@@ -460,12 +401,15 @@ private void mergeBak2Database(String databaseDir) throws IOException, RocksDBEx
...
-    DBInterface checkpointDb = DbTool.getDB(sourceDir, "tmp");
-    byte[] value = sourceDb.get(key);
-    if (isEmptyBytes(value)) {
-      byte[] valueFromTmp = checkpointDb.get(Bytes.concat(simpleEncode(dbName), key));
+    DBInterface checkpointDb = DbTool.getDB(sourceDir, CHECKPOINT_DB);
+    // get data from tmp first.
+    byte[] valueFromTmp = checkpointDb.get(Bytes.concat(simpleEncode(dbName), key));
+    byte[] value;
+    if (isEmptyBytes(valueFromTmp)) {
+      value =  sourceDb.get(key);
```
Comment (guoquanwu) [inline framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java:409]: two blank spaces

File: framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java [hunk 4]
```diff
@@ -500,6 +443,53 @@ private void deleteSnapshotFlag(String databaseDir) throws IOException, RocksDBE
 
   }
 
+  private void hasEnoughBlock(String sourceDir) throws RocksDBException, IOException {
+    // check latest
+    long latest = getLatestBlockHeaderNum(sourceDir);
+    // check first, not 0;
+    long first = 0;
+    DBInterface sourceBlockIndexDb = DbTool.getDB(sourceDir, BLOCK_INDEX_DB_NAME);
+    DBIterator  iterator = sourceBlockIndexDb.iterator();
```
Comment (guoquanwu) [inline framework/src/main/java/org/tron/tool/litefullnode/LiteFullNodeTool.java:452]: two blank space

File: framework/src/test/java/org/tron/program/LiteFullNodeToolTest.java [hunk 5]
```diff
@@ -37,51 +37,48 @@ public class LiteFullNodeToolTest {
...
    */
-  public static void destory(String dbPath) {
-    if (FileUtil.deleteDir(new File(dbPath))) {
-      logger.info("Release resources successful.");
-    } else {
-      logger.info("Release resources failure.");
+  public static void destroy(String dbPath) {
+    File f = new File(dbPath);
+    if (f.exists()) {
+      if (FileUtil.deleteDir(new File(dbPath))) {
```
Comment (guoquanwu) [inline framework/src/test/java/org/tron/program/LiteFullNodeToolTest.java:77]: just delete `f`?

**Review by guoquanwu — APPROVED** [review]

**Pull request merged**